lib/txgh/github_api.rb

```diff
@@ -33,6 +33,24 @@ def create_ref(repo, branch, sha)
       client.create_ref(repo, branch, sha) rescue false
     end
 
+    def update_contents(repo, branch, content_map, message)
+      content_map.each do |path, new_contents|
+        branch = Utils.relative_branch(branch)
+        file = client.contents(repo, { path: path, ref: branch})
+        current_contents = Base64.decode64(file[:content])
+        current_sha = file[:sha]
+
+        new_sha = Utils.git_hash_blob(new_contents)
+        options = { branch: branch }
+
+        if current_sha != new_sha
```

Comment:
Yeah, it would. The thing is, we need to get the original blob SHA anyway because the `update_contents` API requires it (functions as a locking mechanism). I figured we might as well check the SHA since we need it anyway.